### PR TITLE
Enemy pool rework

### PR DESCRIPTION
**Pull request by LiamMillerSE**

### CODE REVIEW

File: Assets/Scripts/Enemies/EnemyPool.cs [hunk 1]
```diff
@@ -5,46 +5,42 @@
...
     Car
 }
-public enum Guns
-{
-    Rifle,
-    Shotgun,
-    PulseRifle,
-    Rangers,
-    Barrett
-}
```
Comment (toryjleo) [inline Assets/Scripts/Enemies/EnemyPool.cs:19]: Please add a quick class comment

File: Assets/Scripts/Enemies/EnemyPool.cs [hunk 2]
```diff
@@ -53,32 +49,32 @@ private void Awake()
...
 
             //instantiate the objects with 
             for(int i =0; i< pool.poolSize; i++)
             {   
-                GameObject obj = Instantiate(pool.prefab); 
-                obj.SetActive(false);
-                Ai objAi = obj.GetComponent<Ai>();
-                objAi.Despawn += ObjAi_Despawn;
+                Ai obj = Instantiate(pool.prefab); 
+                //obj.SetActive(false);
```
Comment (toryjleo) [inline Assets/Scripts/Enemies/EnemyPool.cs:71]: Should we delete these lines?

File: Assets/Scripts/Enemies/VehicleAI.cs [hunk 3]
```diff
@@ -81,13 +79,18 @@ public bool IsConfident()
 
     public override void Init()
     {
+        alive = true;
+        hp = GetComponentInChildren<Health>();
         vehicleController = GetComponent<ArcadeAiVehicleController>();
+        vehicleController.enabled = false;
         DeadEvent += CarDeath;
+        this.Despawn += op_ProcessCompleted;
+        hp.Init(StartingHP);
```
Comment (toryjleo) [inline Assets/Scripts/Enemies/VehicleAI.cs:88]: Is startinghp ever set after it dies and comes back?

File: Assets/Scripts/Enemies/VehicleAI.cs [hunk 4]
```diff
@@ -48,31 +48,29 @@ public override void Attack()
     public override void Update()
     {
         base.Update();
-        Vector3 aimLoc = target.transform.position;
-        aimLoc += target.transform.forward * 10;
-        myGun?.transform.LookAt(aimLoc);
-        if(IsConfident() && Vector3.Distance(transform.position, target.transform.position) <= attackRange)
+        if(movementTarget != null)
```
Comment (toryjleo) [inline Assets/Scripts/Enemies/VehicleAI.cs:51]: nit: 
```suggestion
        if (movementTarget != null)
```

**Review by toryjleo — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Enemies/VehicleAI.cs (hunk 3 above)
Comment (LiamMillerSE) [reply Assets/Scripts/Enemies/VehicleAI.cs:88]: Yes, in the NewLife() function in the base class

**Review by toryjleo — APPROVED** [review]

**Pull request merged**